### PR TITLE
Adding unit tests for Tween

### PR DESCRIPTION
**Pull request by MattTuttle**
Adding unit tests to Tween and finding bugs along the way. One thing that seems strange is that Tween relies on Tweener to complete. That seems like logic that should be kept inside Tween...

Also, removing the macros from the test hxml files lets mcover run. The stats are better than I thought but not great.

### CODE REVIEW

**Comment by bendmorris** [comment]
>Also, removing the macros from the test hxml files lets mcover run.

That's strange. Do they both need to be removed? I recently removed ImportTest (which wasn't up to date) in favor of the --include macro.

**Comment by MattTuttle** [comment]
I had to remove `--include` to get it to work.

**Comment by bendmorris** [comment]
I'd like to find a way to make sure all modules compile without having to maintain a list of all files which quickly gets outdated. Maybe there's a clever way to do that without running a macro up front.

**Comment by MattTuttle** [comment]
I created a new class with a syntax error in it and it failed with the compiler output while running munit. Good idea to make sure that works though!

File: tests/src/haxepunk/BaseTest.hx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package haxepunk;
+
+import massive.munit.Assert;
+
+class BaseTest
```
Comment (bendmorris) [inline tests/src/haxepunk/BaseTest.hx]: Instead of using inheritance, what about providing these as static methods similar to how Assert currently works?

Comment (MattTuttle) [inline tests/src/haxepunk/BaseTest.hx]: I wasn't sure how to handle this. Assert has a `throws` function in the git version of munit but that isn't on haxelib yet.

**Pull request merged**